pkg/tailscale/log.go

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package main
+
+/*
+#cgo LDFLAGS: -llog
+
+#include <stdlib.h>
+#include <android/log.h>
+*/
+import "C"
+
+import (
+	"bufio"
+	"log"
+	"os"
+	"path/filepath"
+	"runtime"
+	"syscall"
+	"unsafe"
+)
+
+// 1024 is the truncation limit from android/log.h, plus a \n.
```

Comment:
the header says implementation defined, "e.g. 1023", though I believe this is closer to 4k (and has been for a very long time), but leaving space for some kernel and userspace header bits, around 4000 bytes in almost all implementations, subtracting whatever tag is set.